gdal/frmts/prf/phprfdataset.cpp

```diff
@@ -0,0 +1,518 @@
+{
+    GDALProxyPoolDataset*   poTileDataset;
+    poTileDataset = new GDALProxyPoolDataset( pszPartName, nWidth, nHeight,
+                                              eAccessType, FALSE );
+
+    if( poTileDataset == NULL )
+    {
+        return -1;
+    }
+
+    for( int nBand = 1; nBand != GetRasterCount() + 1; ++nBand )
+    {
+        PhPrfBand*      poBand = dynamic_cast<PhPrfBand*>( GetRasterBand( nBand ) );
+        GDALRasterBand* poTileBand;
+
+        if( poBand == NULL )
+        {
+            delete poTileDataset;
+            return -1;
+        }
+
+        //! \todo What reason for nBlockXSize&nBlockYSize passed to AddSrcBandDescription
+        poTileDataset->AddSrcBandDescription( poBand->GetRasterDataType(), 0, 0 );
+        poTileBand = poTileDataset->GetRasterBand( nBand );
+
+        if( 0 == nScale )
+        {
+            poBand->AddSimpleSource( poTileBand, 0, 0, nWidth, nHeight,
+                                     nOffsetX, nOffsetY, nWidth, nHeight );
+        }
+        else
+        {
+            poBand->AddOverview( poTileBand );
+        }
+    }
+    return 0;
+}
+
+int PhPrfDataset::Identify( GDALOpenInfo* poOpenInfo )
+{
+    if( EQUAL( CPLGetExtension( poOpenInfo->pszFilename ), PH_PRF_EXT ) )
+    {
+        return TRUE;
+    }
+    else
+    if( EQUAL( CPLGetExtension( poOpenInfo->pszFilename ), PH_DEM_EXT ) )
+    {
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static void GetXmlNameValuePair( const CPLXMLNode* psElt, CPLString& osName,
+                                 CPLString& osValue )
+{
+    for( const CPLXMLNode* psAttr = psElt->psChild; psAttr != NULL; psAttr = psAttr->psNext )
+    {
+        if( psAttr->eType != CXT_Attribute ||
+            psAttr->pszValue == NULL ||
+            psAttr->psChild == NULL ||
+            psAttr->psChild->pszValue == NULL )
+        {
+            continue;
+        }
+        if( EQUAL( psAttr->pszValue, "n" ) )
+        {
+            osName = psAttr->psChild->pszValue;
+        }
+        else
+        if( EQUAL( psAttr->pszValue, "v" ) )
+        {
+            osValue = psAttr->psChild->pszValue;
+        }
+    }
+}
+
+static CPLString GetXmlAttribute( const CPLXMLNode* psElt, const CPLString& osAttrName,
+                                  const CPLString& osDef = CPLString() )
+{
+    for( const CPLXMLNode* psAttr = psElt->psChild; psAttr != NULL; psAttr = psAttr->psNext )
+    {
+        if( psAttr->eType != CXT_Attribute ||
+            psAttr->pszValue == NULL ||
+            psAttr->psChild == NULL ||
+            psAttr->psChild->pszValue == NULL )
+        {
+            continue;
+        }
+        if( EQUAL( psAttr->pszValue, osAttrName ) )
+        {
+            return psAttr->psChild->pszValue;
+        }
+    }
+    return osDef;
+}
+
+static bool ParseGeoref( const CPLXMLNode* psGeorefElt, double* padfGeoTrans )
+{
+    bool                pabOk[6] = {false,false,false,false,false,false};
```

Comment:
Naming convention. Should be abOk since the variable here is the array and not a pointer to the array. Same for papszGeoKeys